packages/smooth_app/lib/pages/scan/continuous_scan_page.dart

```diff
@@ -1,114 +1,140 @@
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: <Widget>[
+                  Padding(
+                    padding: qrScannerPadding,
+                    child: SmoothViewFinder(
+                      boxSize: scannerSize,
+                      lineLength: screenSize.width * 0.8,
                     ),
-                    lineLength: screenSize.width * 0.8,
                   ),
-                ),
-              ],
+                ],
+              ),
             ),
-          ),
-          SmoothRevealAnimation(
-            delay: 400,
-            startOffset: const Offset(0.0, -0.1),
-            animationCurve: Curves.easeInOutBack,
-            child: Column(
-              mainAxisAlignment: MainAxisAlignment.end,
-              children: <Widget>[
-                AnimatedOpacity(
-                  opacity: model.isNotEmpty ? 1.0 : 0.0,
-                  duration: const Duration(milliseconds: 50),
-                  child: Row(
-                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-                    children: <Widget>[
-                      ElevatedButton.icon(
-                        icon: const Icon(Icons.cancel_outlined),
-                        onPressed: model.clearScanSession,
-                        label: const Text('Clear'),
-                      ),
-                      ElevatedButton.icon(
-                        icon: const Icon(Icons.emoji_events_outlined),
-                        onPressed: () => _openPersonalizedRankingPage(context),
-                        label: Text(localizations.myPersonalizedRanking),
-                      ),
-                    ],
-                  ),
-                ),
-                Padding(
-                  padding: const EdgeInsets.symmetric(vertical: 8.0),
-                  child: SmoothProductCarousel(
+            SmoothRevealAnimation(
+              delay: 400,
+              startOffset: const Offset(0.0, -0.1),
+              animationCurve: Curves.easeInOutBack,
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: <Widget>[
+                  _buildButtonsRow(context, model),
+                  const Spacer(),
+                  SmoothProductCarousel(
                     showSearchCard: true,
                     height: carouselHeight,
                   ),
-                ),
-              ],
+                ],
+              ),
+            ),
+          ],
+        ),
+      );
+    });
+  }
+
+  Widget _buildButtonsRow(BuildContext context, ContinuousScanModel model) {
+    final ButtonStyle buttonStyle = ButtonStyle(
+      shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+        RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(18.0),
+        ),
+      ),
+    );
+    return AnimatedOpacity(
+      opacity: areButtonsRendered(model) ? 0.8 : 0.0,
```

Comment:
I guess it would make sense to display the "clear" button when we only have one product.

Reply:
hmm good point, I'm still not sold on this app behavior to be honest, because both buttons are very interdependent in functionality. 

- The Compare button compares all products in scan history
- Clear button clears scan history

So I personally think it makes sense to pair and show them together, but I will bring it up with Pierre, Stephane and others this week